site/src/pages/api/gh-icon/[...data].tsx

```diff
@@ -0,0 +1,100 @@
+import { getData } from 'src/lib/icons';
+import SvgPreview from '../../../components/SvgPreview';
+import { stringify } from 'svgson';
+import { ReactNode } from 'react';
+
+export default async function handler(req, res) {
+  // ReactDOMServer needs to be imported dynamically
+  // https://github.com/vercel/next.js/issues/43810
+  const ReactDOMServer = (await import('react-dom/server')).default;
+
+  const url = req.url.split('/');
+  const data = url.at(-1).slice(0, -4);
+  const src = Buffer.from(data, 'base64')
+    .toString('utf8')
+    .replace(/<svg[^>]*>|<\/svg>/g, '');
+
+  let backdrop: ReactNode;
+  try {
+    if (url.at(-2) !== 'gh-icon') {
+      const backdropString = stringify({
+        type: 'element',
+        name: 'svg',
+        value: undefined,
+        attributes: {},
+        children: (await getData(url.at(-2))).iconNode.map(([name, attributes]) => ({
+          type: 'element',
+          name,
+          attributes,
+          children: undefined,
+          value: undefined,
+        })),
+      }).replace(/<svg[^>]*>|<\/svg>/g, '');
+
+      backdrop = (
+        <>
+          <defs xmlns="http://www.w3.org/2000/svg">
+            <pattern
+              id="pattern"
+              width=".1"
+              height=".1"
+              patternUnits="userSpaceOnUse"
+              patternTransform="rotate(45 50 50)"
+            >
+              <line stroke="red" strokeWidth={0.1} y2={1} />
+              <line stroke="red" strokeWidth={0.1} y2={1} />
+            </pattern>
+          </defs>
+          <mask id="svg-preview-backdrop-mask-outline" maskUnits="userSpaceOnUse">
+            <g stroke="#fff" dangerouslySetInnerHTML={{ __html: backdropString }} />
+            <g dangerouslySetInnerHTML={{ __html: src }} />
+          </mask>
+          <mask id="svg-preview-backdrop-mask-fill" maskUnits="userSpaceOnUse">
+            <g stroke="#fff" dangerouslySetInnerHTML={{ __html: backdropString }} />
+            <g dangerouslySetInnerHTML={{ __html: src }} />
+            <g strokeWidth={1.75} dangerouslySetInnerHTML={{ __html: backdropString }} />
+          </mask>
+          <g
+            strokeWidth={2.25}
+            stroke="url(#pattern)"
+            mask={'url(#svg-preview-backdrop-mask-outline)'}
+          >
+            <rect
+              x="0"
+              y="0"
+              width="24"
+              height="24"
+              fill="url(#pattern)"
+              opacity={0.5}
+              stroke="none"
+            />
```

Comment:
This is pretty edge case, but maybe let's add a second `rect` with no pattern and a lower opacity so that even changes like this will be noticeable:
![image](https://user-images.githubusercontent.com/17746067/232321225-4e4e4fa4-bcc5-48a0-b118-8e970dd81a73.png)

Reply:
I prefer the status quo (left).

<img src="https://user-images.githubusercontent.com/25524993/232322687-a1580ecf-84d0-4159-af04-f36cd0a5959b.svg" width="48%"/> <img src="https://user-images.githubusercontent.com/25524993/232322761-ddf36f6a-938d-4076-8fdf-c40ba11351ea.svg" width="48%"/>

Reply:
Well, yes, out of these two, definitely. But my suggestion would actually be something more like this:
![demo](https://user-images.githubusercontent.com/17746067/232326056-d959506d-7ab9-4329-a166-f25c7a272e08.svg)

Reply:
Ahh, now I get it.